### ISSUE TITLE
TypeError: Cannot destructure property 'configurable' of 'descriptor' as it is undefined.

### ISSUE THREAD

**Issue by sidferreira** [Nozbe/WatermelonDB#1337]
When I try to use my models in jest tests, the use of `@lazy` triggers this error.

Note that I use typescript on my models.

I noticed that `lazy` doesn't use the `makeDecorator` call as other decorators use... may this be the reason?

**Comment by sidferreira** [comment]
https://github.com/Nozbe/WatermelonDB/blob/4e7abb5ccbdccc018d657673ccc2c72ae06d0681/src/decorators/lazy/index.js#L15

https://github.com/Nozbe/WatermelonDB/blob/4e7abb5ccbdccc018d657673ccc2c72ae06d0681/src/decorators/date/index.js#L22

**Comment by castroCrea** [comment]
How to use it then? Did you resolve the issue?

**Comment by sidferreira** [comment]
@castroCrea I switched from ts-jest to babel and it fixed the issue.

**Comment by paul-rchds** [comment]
I am moving to a monorepo and after moving my models to a common package I am getting the same error. The package is compiled with the babel plugins that are mentioned in the installation. Any ideas what could be causing it?

```
{
    "presets": ["@babel/react", "@babel/preset-env"],
    "plugins": [
        ["@babel/plugin-proposal-decorators", { "legacy": true }],
        ["@babel/plugin-proposal-class-properties", { "loose": true }],
        [
            "@babel/plugin-transform-runtime",
            {
                "helpers": true,
                "regenerator": true
            }
        ]
    ],
    "targets": { "esmodules": true }
}
```

**Comment by reyalpsirc** [comment]
Why is this closed? I'm having the same issue with NextJS which uses SWC compiler

**Comment by 1755** [comment]
@reyalpsirc  have you resolved the issue? I've faced the same issue

**Comment by reyalpsirc** [comment]
> @reyalpsirc have you resolved the issue? I've faced the same issue

No, I stopped using Watermelondb with NextJS on the project I was starting.
The only other solution is to avoid using the lazy decorator

**Comment by garygcchiu** [comment]
Still running into this issue when trying to run import my db package containing my watermelondb setup in a node script. Also a monorepo setup. 

```
/node_modules/@nozbe/watermelondb/decorators/lazy/index.js:16
    configurable: configurable,
                  ^
TypeError: Cannot destructure property 'configurable' of 'descriptor' as it is undefined.
    at lazy (/node_modules/@nozbe/watermelondb/decorators/lazy/index.js:16:19)
```

**Comment by garygcchiu** [comment]
For others coming across this issue, I was able to get a workaround by tweaking how I set up my `@lazy` methods. 

Instead of [how the docs suggest](https://watermelondb.dev/docs/Query#custom-queries):
```
@lazy comments = this.collections
		.get<Comment>("comments")
		.query(Q.where("post_id", Q.eq(this.id)));
```

Do:
```
@lazy
get comments(){
     return this.collections
		.get<Comment>("comments")
		.query(Q.where("post_id", Q.eq(this.id)));
}
```

**Comment by garygcchiu** [comment]
For others coming across this issue, I was able to get a workaround by changing how I set up my `@lazy` methods:

Instead of:
```
@lazy commenters = this.collections.get('users').query(
    Q.on('comments', 'post_id', this.id)
)
```

Do:
```
@lazy 
get commenters () {
     return this.collections.get('users').query(
        Q.on('comments', 'post_id', this.id)
    )
}
```

**Comment by isaachinman** [comment]
@sidferreira Can you reopen the issue? Using a specific babel setup to handle decorators doesn't mean this isn't broken.

Been many months. Can we get an answer on this @radex? Would you accept a PR to add `makeDecorator` to `lazy`?
